### ISSUE TITLE
Improve `Style/ReturnNilInPredicateMethodDefinition` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12046]
## 1 case

I have the following code:
```ruby
# frozen_string_literal: true

def foo?
  return true if condition?

  nil
end
```

Rubocop has not detected any offenses:
```ruby
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop file.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```

I expected to receive the following code:

```ruby
# frozen_string_literal: true

def foo?
  return true if condition?

  false
end
```

## 2 case

I have the following code:
```ruby
# frozen_string_literal: true

def foo?
  return true if condition?

  return
end
```

I run rubocop and get offenses:

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop file.rb
Inspecting 1 file
C

Offenses:

file.rb:6:3: C: [Correctable] Style/RedundantReturn: Redundant return detected.
  return
  ^^^^^^
file.rb:6:3: C: [Correctable] Style/ReturnNilInPredicateMethodDefinition: Use return false instead of return in the predicate method.
  return
  ^^^^^^

1 file inspected, 2 offenses detected, 2 offenses autocorrectable
```

but if I run it with autocorrect option I will get the code fixed by first cop (I will get 1 case), not the second cop.

## Rubocop

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.54.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
  - rubocop-thread_safety 0.5.1
```

**Comment by jessepanorama** [comment]
Hey @ydakuka, @koic and @bbatsov, this update can lead to false positives for `ReturnNilInPredicateMethodDefinition`. This example function definition will trigger a violation from rubocop when no violation is actually taking place. This method will never return nil:

```
def hash_has_nil_key?
  a_hash.key?(nil)
end
```

**Comment by koic** [comment]
@jessepanorama That issue may be fixed by https://github.com/rubocop/rubocop/pull/12069. Can you upgrade to RuboCop 1.55.1?

**Comment by jessepanorama** [comment]
Ah, Thanks @koic! sorry for the false alarm. I'll try updating to 1.55.1 and will only bother you again if my false positive case isn't fixed!
